### PR TITLE
initial impl of manual_rfold

### PR DESCRIPTION
**Pull request by asibahi**
Hello,

In an attempt to try and understand how all of this works I tried implementing a basis for a `manual_rfold` lint. I took the logic basically as-is from `manual_next_back`, changing what I could identify. I hope y'all would guide me to a better understanding of what's going on here.

changelog: [`manual_rfold`]: first implementation.

### CODE REVIEW

**Comment by blyxyas** [comment]
Sorry for not replying earlier, I thought that I could review this in time, but I'm busy 
r? rust-lang/clippy

**Comment by blyxyas** [comment]
Oh no, not Manish, he's also occupied.
r? rust-lang/clippy

File: tests/ui/manual_rfold.rs
Comment (blyxyas) [inline tests/ui/manual_rfold.rs:1]: Could you execute `cargo uitest` and then bless the tests? (Generate the `.stderr` files) that's the reason that the CI is failing. This way, we would know if the lint code is doing its job.

Comment (asibahi) [reply tests/ui/manual_rfold.rs:1]: I did and it failed the test. It is one of the things I think I need help with as I don't fully understand.

**Pull request closed without merging**